### ISSUE TITLE
Jukeboxes do not respect transactions

### ISSUE THREAD

**Issue by TropheusJ** [FabricMC/fabric#3680]
using FAPI 0.92.0 on 1.20.1
code:
```
try (Transaction t = Transaction.openOuter()) {
    long inserted = inv.insert(ItemVariant.of(item), item.getCount(), t);
    t.abort();
}
```
before insertion: contains empty stack
![image](https://github.com/FabricMC/fabric/assets/60247969/9e221030-5dff-4b8f-b1be-25d7f9967bbd)
after insert: contains record
![image](https://github.com/FabricMC/fabric/assets/60247969/67baeaed-a826-4180-9858-5798324bd080)
after aborting, this does not get reverted.

**Comment by deirn** [comment]
https://github.com/FabricMC/fabric/pull/3517 fixes this issue in 1.20.4.

**Comment by Syst3ms** [comment]
Shouldn't be too hard to backport.
